Orleans.Providers.MongoDB/StorageProviders/MongoGrainStorage.cs

```diff
@@ -115,9 +106,11 @@ private async Task<T> DoAndLog<T>(string actionName, Func<Task<T>> action)
             }
         }
 
-        protected virtual string ReturnGrainName(string grainType, GrainReference grainReference)
+        protected virtual string ReturnGrainName<T>(string stateName, GrainId grainId, IGrainState<T> grainState)
         {
-            return grainType.Split('.', '+').Last();
+            return stateName == "state"
```

Comment:
Just a heads up here, while looking at the data from the integration test, I noticed employee state was getting stored in `Grainstate` collection.
After some digging, looks like Orleans is returning only "state" instead of the full type name for grains with implicit state definition `Grain<TState>`
This causes state for different grains types with implicit state to be stored in the same collection.